### PR TITLE
feat(krakenfutures): add fetchFundingHistory

### PR DESCRIPTION
**Pull request by rayBastard**
## Summary

Implements `fetchFundingHistory` on top of `/history/account-log`, using its server-side `info=funding rate change` filter (live-verified to actually filter). The account log spells contracts in lower case (`pf_dogeusd`), so the id is uppercased to resolve the market. `count` is only sent when no symbol is given: the endpoint has no contract filter, so a server-side page size would truncate other contracts' rows before the client-side symbol filter runs.

- [x] lint, tsBuild, transpile (Py/PHP/C#/Go/Java) — clean, ruff passes, go build ok
- [x] static fixtures captured live via cli: 1 response (a real hourly funding payment) + 3 request combinations (limit alone, symbol + limit, since + until); request 42/42 + response 20/20 green in JS/Py/PHP
- [x] revert-proofed: dropping the uppercase mapping or sending count with a symbol each fail a fixture
- [x] live smoke: no symbol, symbol + limit, other symbol, since after the payment

## Notes

- The response fixture is captured without a symbol on purpose: with a symbol the caller's market masks the id mapping.
- `accountUid` in the captured body is masked.

### CODE REVIEW

**Review by carlotestor — COMMENTED** [review]
## Summary

`fetchFundingHistory` on `history/account-log` with the server-side `info=funding rate change` filter, `has` flipped to `true`, plus request (3 cases) and response fixtures. Shape follows the `parseIncomes`/`parseIncome` contract, the id is upper-cased for market resolution, and `until` -> `before` / `since` -> `since` mapping is fixture-covered. Reads well overall.

### Findings

1. `parseIncome` mis-attributes unknown contracts when a symbol is passed (`ts/src/krakenfutures.ts:2703`). `safeSymbol (marketId, market)` only falls back to `market` when the id is *not* in `markets_by_id` — so a log row for a contract that is not in the loaded markets (delisted / newly listed / non-instrument entry) is stamped with the caller's symbol and then survives `filterBySymbolSinceLimit`, returning a foreign row as if it belonged to the requested market. Since the contract filter is client-side, this is the exact path that matters here.
2. The `sort` pre-check (`:2624-2630`) is dead. `this.extend (request, params)` already lets a user-supplied `sort` win over `request['sort']`, so reading it back from `params` first changes nothing — `request['sort'] = 'asc';` alone is equivalent and one branch shorter.

Neither is merge-blocking; (1) is worth a line before merge.

Merge gate: 🟢

File: ts/src/krakenfutures.ts [hunk 1]
```diff
@@ -2600,6 +2600,115 @@ export default class krakenfutures extends Exchange {
...
+        //        ...
+        //    }
+        //
+        // the account log spells the contract in lower case, the market ids are upper case
+        const marketId = this.safeStringUpper (income, 'contract');
+        const currencyId = this.safeString (income, 'asset');
+        const timestamp = this.parse8601 (this.safeString (income, 'date'));
+        return {
+            'info': income,
+            'symbol': this.safeSymbol (marketId, market),
```
Comment (carlotestor) [inline ts/src/krakenfutures.ts]: `safeSymbol (marketId, market)` returns `market['symbol']` only when `marketId` is unknown to `markets_by_id`. Because the contract filter here is client-side, a log row for a contract that is not in the loaded markets gets labelled with the caller's symbol and then passes `filterBySymbolSinceLimit`, so `fetchFundingHistory ('DOGE/USD:USD')` can return a foreign contract's payment. Resolve the id without the `market` fallback (pass `undefined` as the second argument) so unmatched rows keep the raw id and get filtered out.

File: ts/src/krakenfutures.ts [hunk 2]
```diff
@@ -2600,6 +2600,115 @@ export default class krakenfutures extends Exchange {
...
+        const request: Dict = {
+            'info': 'funding rate change', // the account log filters by entry type server-side
+        };
+        if (since !== undefined) {
+            request['since'] = since;
+            const sort = this.safeString (params, 'sort');
+            if (sort === undefined) {
+                request['sort'] = 'asc';
+            }
+        }
```
Comment (carlotestor) [inline ts/src/krakenfutures.ts:2624]: The `sort` read-back is a no-op: `this.extend (request, params)` below already lets a user-supplied `sort` override `request['sort']`. `request['sort'] = 'asc';` inside the `since` branch is equivalent with one less local and one less branch.

File: ts/src/krakenfutures.ts (hunk 1 above)
Comment (rayBastard) [reply ts/src/krakenfutures.ts]: Confirmed and fixed — reproduced with a mock first: a pf_delistedusd row came back stamped DOGE/USD:USD on the old code. parseIncome now resolves with this.safeSymbol (marketId) without the market argument, so unknown contracts keep their raw id and get filtered out. No fixture for it: capturing one needs a real row for a contract missing from the loaded markets, and the test account has none, so it's checked in code plus the mock. The same fallback exists in parseTrade (reached via fetchMyTrades, which also filters by symbol on the client side); that parser is shared by several methods, so I'd do it as a separate PR.

File: ts/src/krakenfutures.ts (hunk 2 above)
Comment (rayBastard) [reply ts/src/krakenfutures.ts:2624]: Right — extend already lets the user value win. Set request['sort'] = 'asc' directly, and removed the identical read-back from the sibling fetchLedger; no behaviour change (a user sort: 'desc' still overrides in both, checked with a mock).

**Comment by carlotestor** [comment]
## Merge digest

Adds `fetchFundingHistory` to krakenfutures via `/history/account-log` with the server-side `info=funding rate change` filter.

| File | Δ | What |
| --- | ---: | --- |
| `ts/src/krakenfutures.ts` | +111/−6 | `fetchFundingHistory` |
| `ts/src/test/static/response/krakenfutures.json` | +70/−0 | response fixture |
| `ts/src/test/static/request/krakenfutures.json` | +41/−0 | request fixtures |

```mermaid
flowchart LR
  safeString["safeString"]
  historyGetAccountLog["historyGetAccountLog"]
  parse8601["parse8601"]
  loadMarkets["loadMarkets"]
  krakenfutures["krakenfutures"]
  safeInteger["safeInteger"]
  omit["omit"]
  safeStringUpper["safeStringUpper"]
  extend["extend"]
  parseIncomes["parseIncomes"]
  safeList["safeList"]
  market["market"]
  krakenfutures --> loadMarkets
  krakenfutures --> market
  krakenfutures --> safeInteger
  krakenfutures --> omit
  krakenfutures --> historyGetAccountLog
  krakenfutures --> extend
  krakenfutures --> safeList
  krakenfutures --> parseIncomes
  krakenfutures --> safeStringUpper
  krakenfutures --> safeString
  krakenfutures --> parse8601
```

`ts/src/krakenfutures.ts`

```ts
@@ -2600,6 +2597,114 @@
+    override async fetchFundingHistory (symbol: Str = undefined, since: Int = undefined, limit: Int = undefined, params = {}): Promise<FundingHistory[]> {
+        await this.loadMarkets ();
+        let market: Market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+        }
+        const request: Dict = {
+            'info': 'funding rate change', // the account log filters by entry type server-side
+        };
+        if (since !== undefined) {
+            request['since'] = since;
+            request['sort'] = 'asc';
+        }
```
